builtin/providers/aws/resource_aws_s3_bucket.go

```diff
@@ -201,17 +201,21 @@ func resourceAwsS3BucketCreate(d *schema.ResourceData, meta interface{}) error {
 	// Get the bucket and acl
 	bucket := d.Get("bucket").(string)
 	acl := d.Get("acl").(string)
+	region := d.Get("region").(string)
 
 	log.Printf("[DEBUG] S3 bucket create: %s, ACL: %s", bucket, acl)
 
 	req := &s3.CreateBucketInput{
 		Bucket: aws.String(bucket),
 		ACL:    aws.String(acl),
 	}
-
-	// Special case us-east-1 region and do not set the LocationConstraint.
-	// See "Request Elements: http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUT.html
-	if awsRegion != "us-east-1" {
+	if *aws.String(region) != "" {
+		req.CreateBucketConfiguration = &s3.CreateBucketConfiguration{
+			LocationConstraint: aws.String(region),
+		}
+	} else if awsRegion != "us-east-1" {
+		// Special case us-east-1 region and do not set the LocationConstraint.
+		// See "Request Elements: http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUT.html
```

Comment:
I'm not sure this logic is doing what we'd like... if I set my region to `us-east-1` in the configuration of my S3 Bucket, won't `if *aws.String(region) != "" {` evaluate to `true` and then we'll add a the `LocationConstraint` that we don't want to add?

It does make sense though to read the `region` attribute from the configuration and not just use the default connection's region though, if you'd like to make a change like so:

```go
awsRegion := meta.(*AWSClient).region
// check if region is specified
if v,ok := d.GetOk("region"); ok {
	awsRegion = v.(string)
}

// Special case us-east-1 region and do not set the LocationConstraint.
// See "Request Elements: http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUT.html
if awsRegion != "us-east-1" {
	[...]
}
```

Would that accomplish what you're expecting?